org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/lookup/LookupEnvironment.java

```diff
@@ -110,6 +110,7 @@ public class LookupEnvironment implements ProblemReasons, TypeConstants {
 	private SimpleLookupTable uniqueGetClassMethodBinding; // https://bugs.eclipse.org/bugs/show_bug.cgi?id=300734
 
 	boolean useModuleSystem;					// true when compliance >= 9 and nameEnvironment is module aware
+	boolean ignoreUnnamedModule;
```

Comment:
I'd prefer not to further pollute the state of `LookupEnvironment`, given that the field is just an alias for `this.globalOptions.ignoreUnnamedModuleForSplitPackage`. It's perfectly normal use to ask `environment.globalOptions.xyz` in any client code.